gettsim/transfers/grundrente.py

```diff
@@ -0,0 +1,391 @@
+import numpy as np
+
+from gettsim.typing import BoolSeries
+from gettsim.typing import FloatSeries
+from gettsim.typing import IntSeries
+
+
+def grundr_zuschlag_m(
+    grundr_zuschlag_vor_eink_anr_m: FloatSeries, anrechenbares_eink_gr_m: FloatSeries
+) -> FloatSeries:
+    """Calculate Grundrentenzuschlag (additional monthly pensions payments
+    resulting from Grundrente)
+
+    Parameters
+    ----------
+    grundr_zuschlag_vor_eink_anr_m
+        See :func:`grundr_zuschlag_vor_eink_anr_m`.
+    anrechenbares_eink_gr_m
+        See :func:`anrechenbares_eink_gr_m`.
+
+    Returns
+    -------
+
+    """
+    out = grundr_zuschlag_vor_eink_anr_m - anrechenbares_eink_gr_m
+    return out.clip(lower=0).round(2)
+
+
+def anrechenbares_eink_gr_m(
+    anz_erwachsene_tu: IntSeries,
+    zu_verst_eink_excl_grundr_zuschlag_m_tu: FloatSeries,
+    rentenwert: FloatSeries,
+    tu_id: IntSeries,
+    ges_renten_vers_params: dict,
+) -> FloatSeries:
+    """Calculate reduction of Grundrentenzuschlag through income offsetting.
+
+    Implement income crediting rule as defined in Grundrentengesetz.
+
+    There are upper and lower thresholds for singles and couples. 60% of income between
+    the upper and lower threshold is credited against the Grundrentenzuschlag. All the
+    income above the upper threshold is credited against the Grundrentenzuschlag.
+
+    Parameters
+    ----------
+    anz_erwachsene_tu
+        See :func:`anz_erwachsene_tu`.
+    zu_verst_eink_excl_grundr_zuschlag_m_tu
+        See :func:`zu_verst_eink_excl_grundr_zuschlag_m_tu`.
+    rentenwert
+        See :func:`rentenwert`.
+    tu_id
+        See basic input variable :ref:`tu_id <tu_id>`.
+    ges_renten_vers_params
+        See params documentation :ref:`ges_renten_vers_params <ges_renten_vers_params>`.
+    Returns
+    -------
+
+    """
+    alleinstehend_grundr = tu_id.replace(anz_erwachsene_tu < 2)
```

Comment:
Shouldn't we have that as a separate function? Maybe we already check if tax units have a single or multiple persons?! If not it should be in demographic vars.